pkg/workflows/management/create_cluster.go

```diff
@@ -0,0 +1,379 @@
+package management
+
+import (
+	"context"
+
+	"github.com/aws/eks-anywhere/pkg/clustermarshaller"
+	"github.com/aws/eks-anywhere/pkg/constants"
+	"github.com/aws/eks-anywhere/pkg/logger"
+	"github.com/aws/eks-anywhere/pkg/task"
+	"github.com/aws/eks-anywhere/pkg/types"
+	"github.com/aws/eks-anywhere/pkg/workflows"
+)
+
+type createBootStrapClusterTask struct{}
+
+type createWorkloadClusterTask struct{}
+
+type installResourcesOnManagementTask struct{}
+
+type installEksaComponentsTask struct{}
+
+type installEksaComponentsOnWorkloadTask struct{}
+
+type installGitOpsManagerTask struct{}
+
+type moveClusterManagementTask struct{}
+
+type writeClusterConfigTask struct{}
+
+type deleteBootstrapClusterTask struct{}
+
+type installCuratedPackagesTask struct{}
+
+func (s *createBootStrapClusterTask) Run(ctx context.Context, commandContext *task.CommandContext) task.Task {
+	logger.Info("Creating new bootstrap cluster")
+
+	bootstrapOptions, err := commandContext.Provider.BootstrapClusterOpts(commandContext.ClusterSpec)
+	if err != nil {
+		commandContext.SetError(err)
+		return nil
+	}
+
+	bootstrapCluster, err := commandContext.Bootstrapper.CreateBootstrapCluster(ctx, commandContext.ClusterSpec, bootstrapOptions...)
+	if err != nil {
+		commandContext.SetError(err)
+		return nil
+	}
+	commandContext.BootstrapCluster = bootstrapCluster
+
+	logger.Info("Provider specific pre-capi-install-setup on bootstrap cluster")
+	if err = commandContext.Provider.PreCAPIInstallOnBootstrap(ctx, bootstrapCluster, commandContext.ClusterSpec); err != nil {
+		commandContext.SetError(err)
+		return &workflows.CollectMgmtClusterDiagnosticsTask{}
+	}
+
+	logger.Info("Installing cluster-api providers on bootstrap cluster")
+	if err = commandContext.ClusterManager.InstallCAPI(ctx, commandContext.ClusterSpec, bootstrapCluster, commandContext.Provider); err != nil {
```

Comment:
why is capi in the create bootstrap cluster task? and if we want capi here, why isn't eksa here as well?
trying to understand the reasoning here

Reply:
So CAPI was always in the create bootstrap cluster task, so I left it there itself. I created a new one below for installing the eksa components tasks, but if you have a strong preference I can combine them all into the create bootstrap cluster task.